cmd/sdkstats/main.go

```diff
@@ -7,46 +7,120 @@ import (
-	queries = append(queries, "filename:Dockerfile quay.io/operator-framework/ansible-operator")
-	queries = append(queries, "filename:go.mod github.com/operator-framework/operator-sdk")
-	queries = append(queries, "filename:Gopkg.toml github.com/operator-framework/operator-sdk")
-
+	queries := []sdkstats.RepoMetadataQuery{
+		sdkstats.RepoMetadataQuery{
+			ProjectType:   "ansible",
+			Queries:       []string{"filename:Dockerfile quay.io/operator-framework/ansible-operator"},
+			VersionParser: &projectVersionParser{},
+		},
+		sdkstats.RepoMetadataQuery{
+			ProjectType:   "helm",
+			Queries:       []string{"filename:Dockerfile quay.io/operator-framework/helm-operator"},
+			VersionParser: &projectVersionParser{},
+		},
+		sdkstats.RepoMetadataQuery{
+			ProjectType: "go",
+			Queries: []string{
+				"filename:go.mod github.com/operator-framework/operator-sdk",
+				"filename:Gopkg.toml github.com/operator-framework/operator-sdk",
+			},
+			VersionParser: &projectVersionParser{},
+		},
+	}
+	//GetStats function for Query String from 'queries', These Strings are specific to Operator-SDK patterns.
 	for _, r := range queries {
 		stats, err := sdkstats.GetStats(client, r)
-		if len(stats) == 0 {
-			log.Fatal("GITHUB API is down at time, Please try after few minutes")
-		} else {
-			fileName := stats[0].ProjectType + ".json"
-			if err != nil {
-				fmt.Println(err)
-			}
-			fmt.Println("Total count: ", len(stats))
-			file, _ := json.MarshalIndent(stats, "", " ")
-			_ = ioutil.WriteFile(fileName, file, 0644)
-			fmt.Println("Results are written in ", fileName)
+		if _, ok := err.(*github.AcceptedError); ok {
+			log.Println("Job is scheduled on GitHub side")
+		} else if _, ok := err.(*github.RateLimitError); ok {
+			log.Println("Rate Limit has reached.")
+		} else if err != nil {
+			fmt.Println("Failed to get Stats for Query", r, err)
 		}
+		//Write results into JSON , named with ProjectType.
+		fileName := r.ProjectType + ".json"
+		fmt.Println("Total count: ", len(stats))
+		file, _ := json.MarshalIndent(stats, "", " ")
+		_ = ioutil.WriteFile(fileName, file, 0644)
+		fmt.Println("Results are written in ", fileName)
 	}
 
 }
+
+//Parse the given Code result to search Text Matches for Version number.
+func (p projectVersionParser) ParseVersion(codeResults github.CodeResult, projectType string) (string, error) {
+	var searchQ, sdkVersion, searchLatest string
+	sdkLatest := "11"
+	var posFirstAdjusted, c int
+	//Text match strings for Helm/Ansible are always from the First line of Docker file, Hence fixing the Version Indices.
+	if projectType == "helm" {
+		searchQ = "quay.io/operator-framework/helm-operator:v0."
+		searchLatest = "quay.io/operator-framework/helm-operator:latest"
+		posFirstAdjusted = 49
+		c = 51
+	} else if projectType == "ansible" {
+		searchQ = "quay.io/operator-framework/ansible-operator:v0."
+		searchLatest = "quay.io/operator-framework/ansible-operator:master"
```

Comment:
I did put some logic earlier to grab everything after "v0.", However, not all CodeSearch results in GO based projects had a defined separator, like /n. However, This worked for "helm", and "ansible". But, I wanted to keep logic similar across Project Types, hence let go of that thought process. WDYT

Reply:
Ah, right, but with the context of [this comment](https://github.com/learnoperators/go-github-metrics/pull/12#discussion_r341359312) in mind, I think we can have separate VersionParser implementations to handle each of the result formats that we expect to find from our queries:

```go
// Used for helm and ansible queries
type baseImageVersionParser struct {
    baseImage string
}
func (p baseImageVersionParser) ParseVersion(result github.CodeResult) (string, error)
    // find p.baseImage + ":" in the result, and return everything to the right of it
}

// Used for go.mod queries
type goModVersionParser struct {
    module string
}
func (p goModVersionParser) ParseVersion(result github.CodeResult) (string, error)
    // return strings.Strip(<everything after p.module>)
}

// Used for Gopkg.toml queries
type unknownVersionParser struct {}
func (_ unknownVersionParser) ParseVersion(_ github.CodeResult) (string, error)
    return "unknown", nil
}
```

It may be more complicated than that, but that's the gist of what I'm suggesting.

Reply:
Yes, I am okay with this change, Will make it.